server/internal/application/user/user.go

```diff
@@ -0,0 +1,62 @@
+package user
+
+import (
+	"errors"
+	"time"
+)
+
+var (
+	ErrNameNotValid = errors.New("pippo is not valid")
+)
+
+type User struct {
+	Username  string
+	Password  string
+	Name      string
+	Surname   string
+	Email     string
+	Role      string
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
```

Comment:
- we realistically only need the username and the timestamps for the MVP 
- we don't have roles for now as we won't have a way to add more users
- the User object should not hold the clear text password, but rather the password hash
- if we decide to add a display name in the future, splitting between first name and surname is not inclusive: many cultures have a middle name (e.g. Anglo-Saxons, French), some have multiple surnames (e.g. Spanish and Latinx countries), some use surnames differently (e.g. Chinese and Japanese), and some don't have the concept of a surname at all! Best just have a single `name` field
- the User aggregate should hold the domain logic for manipulating the data it contains, but here's it's just a plain struct